src/compiler/checker.ts

```diff
@@ -13328,8 +13328,10 @@ namespace ts {
                 case SyntaxKind.TemplateSpan:
                     Debug.assert(parent.parent.kind === SyntaxKind.TemplateExpression);
                     return getContextualTypeForSubstitutionExpression(<TemplateExpression>parent.parent, node);
-                case SyntaxKind.ParenthesizedExpression:
-                    return getContextualType(<ParenthesizedExpression>parent);
+                case SyntaxKind.ParenthesizedExpression: {
+                    const tag = isInJavaScriptFile(parent) ? getJSDocTypeTag(parent) : undefined;

```

Comment:
I think a one-line comment would help remind the reader why parentheses might be special in a js file:

"Parentheses could be part of the JSDoc cast syntax invented by Closure."
Or similar.